### PR TITLE
HHH-12939 Properly quote identifiers in alter table statements

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HHH-12939

This is a follow-up of https://github.com/hibernate/hibernate-orm/pull/2512, which is IMHO the proper fix for the original https://hibernate.atlassian.net/browse/HHH-12106 issue.

Added 2 test cases run with H2.

@vladmihalcea apart from a review, if you have SQL Server handy, I would appreciate if you could check that the modified SQL Server test works as expected.

@pbruski feedback welcome.

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/test/schemaupdate/AlterTableQuoteDefaultSchemaTest.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.hibernate.tool.hbm2ddl.SchemaUpdate;
+import org.hibernate.tool.schema.TargetType;
+import org.junit.Test;
+
+/**
+ * @author Guillaume Smet
+ */
+@TestForIssue(jiraKey = "HHH-12939")
+@RequiresDialect(H2Dialect.class)
```
Comment (vladmihalcea) [inline hibernate-core/src/test/java/org/hibernate/test/schemaupdate/AlterTableQuoteDefaultSchemaTest.java]: Better use:

    @RequiresDialectFeature(DialectChecks.SupportSchemaCreation.class)

So we can test it on more DBs, like SQL Server.

File: hibernate-core/src/test/java/org/hibernate/test/schemaupdate/AlterTableQuoteSpecifiedSchemaTest.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.hibernate.tool.hbm2ddl.SchemaUpdate;
+import org.hibernate.tool.schema.TargetType;
+import org.junit.Test;
+
+/**
+ * @author Guillaume Smet
+ */
+@TestForIssue(jiraKey = "HHH-12939")
+@RequiresDialect(H2Dialect.class)
```
Comment (vladmihalcea) [inline hibernate-core/src/test/java/org/hibernate/test/schemaupdate/AlterTableQuoteSpecifiedSchemaTest.java]: Better use:

    @RequiresDialectFeature(DialectChecks.SupportSchemaCreation.class)

So we can test it on more DBs, like SQL Server.

File: hibernate-core/src/test/java/org/hibernate/test/schemaupdate/AlterTableQuoteDefaultSchemaTest.java (hunk 1 above)
Comment (gsmet) [reply hibernate-core/src/test/java/org/hibernate/test/schemaupdate/AlterTableQuoteDefaultSchemaTest.java]: That's exactly what I don't want to do: quoting rules are different for each dialect so I just want to validate the behavior works for one dialect supporting quoting.

**Comment by gsmet** [comment]
The failure on SQL Server is expected as it's not quoting strings the same way as H2.

I wasn't clear but the test to validate with SQL Server is this one: `SqlServerQuoteSchemaTest`. I adapted it so I expect it to work but better be sure.

**Comment by vladmihalcea** [comment]
I can send a PR based on yours with the test adapted for all major DBs supporting schemas. I need to test it on PostgreSQL too.

**Comment by gsmet** [comment]
If you want but I don't think it has much value as we just want to validate the general mechanism.

But feel free to do it.

**Comment by gsmet** [comment]
Superseded by https://github.com/hibernate/hibernate-orm/pull/2528.

**Pull request closed without merging**